pandas/_libs/lib.pyx

```diff
@@ -1882,7 +1889,7 @@ cdef class StringValidator(Validator):
 
     cdef bint is_valid_null(self, object value) except -1:
         # We deliberately exclude None / NaN here since StringArray uses NA
-        return value is C_NA
+        return value is C_NA or value is None or util.is_nan(value)
```

Comment:
Quick question: 
If I do ``pd.arrays.StringArray(np.array(["a", np.nan], dtype=object))``, won't this result in a StringArray with ``np.nan`` in it? (something like this? https://github.com/pandas-dev/pandas/issues/30966#issuecomment-573698662)

Reply:
> won't this result in a StringArray with np.nan in it?

yep.  probably ought to do something about that.  have something min mind?

Reply:
I'm trying to fix this in #41412. This PR is probably stuck in the meantime.

Reply:
@lithomas1 is this un-blocked?